src/test/java/seedu/address/model/event/LocationTest.java

```diff
@@ -22,39 +22,44 @@ void constructor_null_throwsNullPointerException() {
 
     @Test
     void constructor_invalidLocation_throwsIllegalValueException() {
+        // whitespace -> throws IllegalArgumentException
         assertThrows(IllegalArgumentException.class, () -> new Location(""));
         assertThrows(IllegalArgumentException.class, () -> new Location("  "));
         assertThrows(IllegalArgumentException.class, () -> new Location("     "));
 
+        // non-alphanumeric -> throws IllegalArgumentException
+        assertThrows(IllegalArgumentException.class, () -> new Location(LOCATION_WITH_CAROT));
+        assertThrows(IllegalArgumentException.class, () -> new Location(LOCATION_WITH_AT));
+        assertThrows(IllegalArgumentException.class, () -> new Location(LOCATION_WITH_SLASH));
+
         // inputs with leading spaces are rejected
         assertThrows(IllegalArgumentException.class, () -> new Location(LOCATION_WITH_LEADING_SPACES));
     }
 
     @Test
     void constructor_validLocation_success() {
-        assertNotNull(new Location(LOCATION_WITH_CAROT));
-        assertNotNull(new Location(LOCATION_WITH_AT));
-        assertNotNull(new Location(LOCATION_WITH_SLASH));
-        assertNotNull(new Location(LOCATION_WITH_TRAILING_SPACES));
+        assertNotNull(new Location(VALID_LOCATION));
     }
 
     @Test
     void isValidLocation() {
```

Comment:
Should there be some positive test cases as well?
Similar for `EventNameTest`.

Reply:
Good point. Fixed.